examples/linux64/file-reading/reading_file.c

```diff
@@ -0,0 +1,73 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#define	O_RDONLY	0x0000		/* open for reading only */
+#define	O_WRONLY	0x0001		/* open for writing only */
+#define	O_RDWR		0x0002		/* open for reading and writing */
+#define	O_ACCMODE	0x0003		/* mask for above modes */
+
+inline size_t read(int fd, void *buf, size_t count){
+	return (size_t) syscall(0, fd, buf, count);
+}
+
+inline int open(const char *filename, int mode){
+	return syscall(2, filename, mode, 0);
+}
+
+inline int close(int fd){
+    return syscall(3, fd, 0, 0);
+}
+
+inline void fstat(int fd, void * statstruct){
+    syscall(5, fd, (long) statstruct, 0);
+}
+
+inline int size(int fd){
+	// needs to inform the size of kernel linux stat struct size
+	// which is 144 bytes in x86_64 systems
+    char buffer[144];
+
+    void * statstruct = (void*)(&buffer);
+
+    fstat(fd, statstruct);
+    
+    // from the begin to the size part, we have 
+	// 8 + 8 + 8 + 8 + 4 + 4 + 8 = 48 bytes
+	long size = *((long*)(statstruct + 48));
+
+    return size;
+}
+
+int main(int argc, char **argv){
+    int fd = open("example.txt",O_RDONLY);
```

Comment:
Ah, I understand the purpose of example.txt now :). Cool that this example works!!

Reply:
So I dont need to replace the .txt for the README, but it would be nice to replace the content with a simple description of the example, as you said in a previous comment, right?

Reply:
Yeah, I guess so, or you could still add a README.md and read this file with the C example code. This makes for some inception like effect :D.